scripts/o3de/o3de/gitproviderinterface.py

```diff
@@ -28,9 +28,23 @@ def clone_from_git(self, uri: ParseResult, download_path: pathlib.Path, force_ov
         """
         Clones a git repository from a uri into a given folder
         :param uri: uniform resource identifier
-        :param download_path: location path on disk to download file
+        :param download_path: location path on disk to download files
         :param force_overwrite: whether to force overwrite the contents that already exist on disk or not
         :param ref: optional source control reference which can be a commit hash, branch, tag or other reference type
         :return: return code, 0 on success
         """
         pass
+    
+    @abstractmethod
+    def upload_release_to_github(self, repo_uri: ParseResult, zip_path: pathlib.Path, archive_filename: str, upload_git_release_tag: str):
```

Comment:
Lets not use `github` in the function name so it still makes sense when/if we add other git providers
```suggestion
    def upload_release(self, repo_uri: ParseResult, zip_path: pathlib.Path, archive_filename: str, git_release_tag: str):
```